### DISCUSSION TITLE
Components renaming in 2.10.0

### DISCUSSION THREAD

**Discussion by lubber-de** [fomantic/Fomantic-UI#2407] [General]
We plan/thought about to rename wrong terminology of component names in 2.10.0 (Relax, this will take time...)

For now it will only affect the `header` component, which will become a `heading` instead
https://github.com/Semantic-Org/Semantic-UI/issues/7077
https://github.com/Semantic-Org/Semantic-UI/issues/5972

We also plan to prepare a switch variable like `@variationHeaderAlias:true` to also provide the old name for every selector at the same time for easier upgrading, so existing code should still work.
However, this upgrade support will double all of the head(er/ing) selectors so we will disable the alias feature by default possibly by 2.11.0 (Again, relax, this will take much more time)

Please share your feedback about this. Do you think other components should be renamed as well?

**Comment by hugopeek** [comment]
100% agree with the name change for header! Go for it :rocket: 

We've talked about the double 'form' component before, both as collection and behaviour: https://github.com/fomantic/Fomantic-UI/pull/2209

And there are a few inconsistencies inside components, like different names for padding (very padded / very relaxed). Nice if we could align those, but it will take some digging to find them and a bit of work too to keep it backwards compatible.

**Comment by mvorisek** [comment]
I also propose to rename the `semantic.js`/css to `fomantic.js`/css and all other occurences\* of "semantic" (but keep the mention that Fomantic-UI is fork of Semantic-UI of course).

Currently the mixed naming of semantic/fomantic is very confusing.

\* https://github.com/fomantic/Fomantic-UI/search?q=semantic

---

also rename `$.fn.form.settings.rules.empty` to `$.fn.form.settings.rules.notEmpty` -> issue: https://github.com/fomantic/Fomantic-UI/issues/2937

**Comment by lubber-de** [comment]
As suggested in https://github.com/Semantic-Org/Semantic-UI/issues/3275
We may rename `tabular` into `tabbed` menu ?

**Comment by lubber-de** [comment]
Rename `header` to `heading` is now prepared by #3200
You may read my PR description for some more thoughts about this, especially how to prepare users/existing projects for an easy/quick migration as this really is a huge breaking change if it once comes unexpected

**Comment by lubber-de** [comment]
Rename `tabular` to `tabbed` is now prepared by #3201
